open-feature/php-sdk
# feat!: update minimum supported PHP version to 8.1

## This PR

Updates the minimum supported version of the SDK to PHP 8.1
Introduces new PHP language features including readonly properties, enumerations, 

- minimum supported version of PHP is now 8.1
- refactors pseudo-enums to true enums
- utilizes readonly properties where applicable
- utilizes array unpacking, e.g. `[...$array]` over `array_merge($array)`

### Related Issues

Resolves #108

### Follow-up Tasks

General TODOs before leaving `DRAFT`:

- [x] Resolve refactor failures in tests and static analysis
- [ ] Update flagd provider to support breaking changes from major version release

### How to test

Locally, you can run `composer run test` to execute the full test suite

Remotely, see GitHub CI runs
